src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -0,0 +1,328 @@
+          (scaling_and_sphericity + trigonometric_nonsphericity) /
+          denominator_other;
+  dxdeta = -scaling_and_sphericity * sy_square * y_equiangular_tangent *
+           x_equiangular_tangent / denominator_other;
+  dxdzeta = (nonsphericity * xi +
+             (-inner_sphericity_ + ratio_of_radii) * x_equiangular_tangent) /
+            denominator_dzeta;
+  dydxi = -scaling_and_sphericity * sx_square * y_equiangular_tangent *
+          x_equiangular_tangent / denominator_other;
+  dydeta = sy_square * sx_square *
+           (scaling_and_sphericity + trigonometric_nonsphericity) /
+           denominator_other;
+  dydzeta = (nonsphericity * eta +
+             (-inner_sphericity_ + ratio_of_radii) * y_equiangular_tangent) /
+            denominator_dzeta;
+  dzdxi = -scaling_and_sphericity * sx_square * x_equiangular_tangent /
+          denominator_other;
+  dzdeta = -scaling_and_sphericity * sy_square * y_equiangular_tangent /
+           denominator_other;
+  dzdzeta =
+      (nonsphericity - inner_sphericity_ + ratio_of_radii) / denominator_dzeta;
+
+  // Implement Rotation:
+  if (positioning_of_wedge_ == Direction<3>::lower_zeta()) {
+    dzdxi *= -1;
+    dzdeta *= -1;
+    dzdzeta *= -1;
+
+    dydxi *= -1;
+    dydeta *= -1;
+    dydzeta *= -1;
+  }
+
+  if (positioning_of_wedge_.axis() == Direction<3>::Axis::Eta) {
+    std::swap(dxdxi, dydxi);
+    std::swap(dxdeta, dydeta);
+    std::swap(dxdzeta, dydzeta);
+    std::swap(dydxi, dzdxi);
+    std::swap(dydeta, dzdeta);
+    std::swap(dydzeta, dzdzeta);
+
+    if (positioning_of_wedge_.side() == Side::Lower) {
+      dydxi *= -1;
+      dydeta *= -1;
+      dydzeta *= -1;
+
+      dzdxi *= -1;
+      dzdeta *= -1;
+      dzdzeta *= -1;
+    }
+  }
+
+  if (positioning_of_wedge_.axis() == Direction<3>::Axis::Xi) {
+    std::swap(dxdxi, dzdxi);
+    std::swap(dxdeta, dzdeta);
+    std::swap(dxdzeta, dzdzeta);
+    std::swap(dzdxi, dydxi);
+    std::swap(dzdeta, dydeta);
+    std::swap(dzdzeta, dydzeta);
+
+    if (positioning_of_wedge_.side() == Side::Lower) {
+      dxdxi *= -1;
+      dxdeta *= -1;
+      dxdzeta *= -1;
+
+      dydxi *= -1;
+      dydeta *= -1;
+      dydzeta *= -1;
+    }
+  }
+
+  const double scaled_inner_radius = inner_radius_ / sqrt(3.0);
+
+  // multiply jacobian_matrix by scaling factor to get jacobian.
+  std::transform(jacobian_matrix.begin(), jacobian_matrix.end(),
+                 jacobian_matrix.begin(),
+                 [scaled_inner_radius](const T& jacobian_element) {
+                   return jacobian_element * scaled_inner_radius;
+                 });
+  return jacobian_matrix;
+}
+
+template <typename T>
+Tensor<std::decay_t<tt::remove_reference_wrapper_t<T>>,
+       tmpl::integral_list<std::int32_t, 2, 1>,
+       index_list<SpatialIndex<3, UpLo::Up, Frame::NoFrame>,
+                  SpatialIndex<3, UpLo::Lo, Frame::NoFrame>>>
+Wedge3D::inv_jacobian(const std::array<T, 3>& xi) const {
+  const auto jac = jacobian(xi);
```

Comment:
Don't move const things, and here just inlining it into the next expression seems clearer anyway.